### ISSUE TITLE
Remove Azure CLI and other uneeded packages

### ISSUE THREAD

**Issue by Cryptophobia** [teamhephy/docker-go-dev#4]
We should remove the Azure CLI and other unneeded packages especially since we are only using this image to run golang tests and linters mostly.

**Comment by felixbuenemann** [comment]
Fully agree, most of the later additions to the go-dev image are from times where the deis team was already working for Microsoft. 

I haven't looked at all the additions, but working from the official golang docker images would also be an option. 

It would also be worth to review how much deis base image brings to the table compared to plain ubuntu images, which have become Very lightweight in recent versions. Unless we always add the same stuff to the base image for multiple apps and there is no automation to rebuild all apps regularly on base image updates there isn't much benefit to using it.

Also since the base image is still using Ubuntu Xenial and Focal is the latest release, it might be time to upgrade. There is no immediate pressure, since Xenial is EOL in 2021 if I remember correctly, but a lot has changed since then.

**Comment by Cryptophobia** [comment]
Yes, I agree with everything @felixbuenemann . Also it's a bit complicated that there are two base images that we are using right now. One is the go-dev image (contains all of the golang development tooling inside) - https://github.com/teamhephy/docker-go-dev . Mostly the router and the builder use that one.

There is also the docker-base image that I think we updated a bit (@kingdonb updated in the past) but haven't been making release of on the regular at https://github.com/teamhephy/docker-base . Also some of the components are still pulling older tags of it. 

We should probably do what bacongobbler mentions here and create a latest tag on the docker-base image so that we can reference all components to pull the latest of the hephy/base when building: https://github.com/teamhephy/docker-base/issues/1
